source/git/gitsource.go

```diff
@@ -686,3 +694,17 @@ func tokenScope(remote string) string {
 	}
 	return remote
 }
+
+// getDefaultBranch gets the default branch of a repository using ls-remote
+func getDefaultBranch(ctx context.Context, gitDir, workDir, sshAuthSock, knownHosts string, auth []string, remoteURL string) (string, error) {
+	buf, err := gitWithinDir(ctx, gitDir, workDir, sshAuthSock, knownHosts, auth, "ls-remote", "--symref", remoteURL, "HEAD")
+	if err != nil {
+		return "", errors.Wrapf(err, "error fetching default branch for repository %s", redactCredentials(remoteURL))
```

Comment:
Can we fallback to "master" on error?

Reply:
Do you know a case that wouldn't work? The problem is that this is the first network request via git so there is actually a good chance that this will legitimately fail and I wouldn't want to hide the error.

Reply:
When was `ls-remote` introduced?
Some git server implementation may not support `ls-remote`?

Reply:
We already had a dependency on `ls-remote` before to get the sha for the ref.